etc/scripts/licenses/report_license_rules.py

```diff
@@ -0,0 +1,198 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ScanCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/scancode-toolkit for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import io
+
+import click
+import csv
+
+from commoncode.cliutils import PluggableCommandLineOption
+from licensedcode.models import load_licenses
+from licensedcode.models import load_rules
+
+
+LICENSES_FIELDNAMES = [
+    'key', 'short_name', 'name', 'category', 'owner',  'text', 'words_count', 'notes',
+    'minimum_coverage', 'homepage_url', 'is_exception', 'language', 'is_unknown',
+    'spdx_license_key', 'text_urls', 'other_urls', 'standard_notice',
+    'license_filename', 'faq_url', 'ignorable_authors',
+    'ignorable_copyrights', 'ignorable_holders', 'ignorable_urls', 'ignorable_emails',
+    'osi_license_key', 'osi_url', 'other_spdx_license_keys',
+]
+
+
+RULES_FIELDNAMES = [
+    'identifier', 'license_expression', 'relevance', 'text', 'words_count', 'category',
+    'is_false_positive', 'is_license_text', 'is_license_notice',  'is_license_tag',
+    'is_license_reference', 'is_license_intro', 'has_unknown', 'only_known_words',
+    'notes', 'referenced_filenames', 'minimum_coverage', 'ignorable_copyrights',
+    'ignorable_holders', 'ignorable_authors', 'ignorable_urls', 'ignorable_emails',   
+]
+
+
+def write_data_to_csv(data, output_csv, fieldnames):
+    
+    with open(output_csv,'w',encoding='utf-8-sig',newline='') as f:
+        w = csv.DictWriter(f,fieldnames=fieldnames)
+        w.writeheader()
+
+        for entry in data:
+            w.writerow(entry)
+
+
+def filter_by_attribute(data, attribute, expected):
+    
+    filtered_output = []
+    for entry in data:
+        if entry.get(attribute, 'None') == expected:
+            filtered_output.append(entry)
+            
+    return filtered_output
+
+def flatten_output(data):
+
+    if not isinstance(data, list):
+        return
```

Comment:
say I have a tuple or a dict?... the flatten_output is then empty? The implicit exclusion of non-list is a bit surprising. If you only ever expect a list, then instead use `assert isinstance(data, list)`